### DISCUSSION TITLE
Can goal_state get velocities?

### DISCUSSION THREAD

**Discussion by Sepirence** [NVlabs/curobo#422] [Software Q&A] [answered]
I am trying to use plan_single_js(). I thought that it can have exit velocities since JointState has velocity field, but generation looks like stop on the goal_state. I found that start_state can get velocities or accelerations if optimize_dt is false, but I could not find the reference of exit velocities. How can I solve it?
Or, I am trying to plan smooth trajectory with some waypoints. Is there any way to make smooth trajectory without planning exit velocity?

**Comment by balakumar-s** [comment, marked as answer]
Goal state only supports position and it assumes that the robot will be static at the final time step. In the current implementation, tracking goal velocities will require significant changes.
